### PR TITLE
add docs about networking requirements

### LINKED ISSUE #93: Clarify networking requirements
The threshold-ECDSA protocol requires two types of communication channels: point-to-point and broadcast. The tss-ecdsa library does not (and should not!) implement any networking; it returns messages after each protocol round that the calling application is responsible for sending over a point-to-point channel. It has special support for broadcast and implements an echo broadcast functionality (again, doing the checks but not the networking).

Goals of this epic:
- [x] Document networking requirements at the crate level, so calling applications understand what they need to provide to have a correct protocol implementation.
- [x] Eliminating this task from this epic; we'll make a separate epic when we're ready to address it. ~Reconsider the broadcast API. A valid broadcast channel is a security-critical element of the protocol, which is likely why a specific implementation was included in the tss-ecdsa library. However, since we use this library as part of a larger protocol, it may be reasonable to leave it up to the calling application to determine how to implement the broadcast channel. If we take this approach, we should carefully document what a cryptographically-secure broadcast channel is and provide references on how to implement it. We can also consider tagging output messages more carefully to indicate what kind of channel they should be sent over.~

### LINKED ISSUE #294: Add documentation about networking requirements for calling application
The calling application has to ensure correct instantiation of all networking code in order to have a secure protocol run. A lot of these requirements are noted in the boltlabs-inc/LK-crypto-protocols repo, but they should be explained in this crate as well. Add explicit documentation about this, including:

- [ ] Describe the security requirements on point-to-point channels
- [ ] Describe the security requirements of the broadcast channel and how the library currently handles it (echo broadcast)
- [ ] Talk about message validation and participant IDs (see [Alex's comment](https://github.com/boltlabs-inc/tss-ecdsa/pull/278#discussion_r1165920807), [swan's comment](https://github.com/boltlabs-inc/tss-ecdsa/pull/279#discussion_r1165717682))
- [ ] Consider mapping from "real-world entities" to participant IDs ([see swan's comment](https://github.com/boltlabs-inc/tss-ecdsa/pull/279#discussion_r1168932161))

### PR DESCRIPTION
**Pull request by marsella**
Closes #294.

Requesting @indomitableSwan as a reviewer. @gatoWololo might also take a pass for whether this makes sense.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -42,23 +42,23 @@
...
-//! the 🔒 symbol.
+//! the 🔒 symbol. At a high level, they include:
 //!
-//! 1. Networking. The protocol requires point-to-point channels between
-//! individual participants, as well as a UC-secure, synchronous broadcast
-//! mechanism. This library currently supports broadcast using the
-//! echo-broadcast protocol described by Goldwasser and Lindell[^echo], so the
-//! calling application only has to implement point-to-point channels.
-//! This may change in future library versions.
+//! 1. Networking. The protocol requires messages to be sent over secure
```
Comment (indomitableSwan) [inline src/lib.rs:52]: I would just move up the details to this section. L52-53 will just confuse people.

File: src/lib.rs [hunk 2]
```diff
@@ -88,6 +88,37 @@
 //! At this time, we do not recommend use for security-critical applications.
 //! Use at your own risk.
 //!
+//! # 🔒 Networking
+//!
+//! The calling application is responsible for sending messages between
+//! participants in the protocol.
+//! All communication between [`Participant`]s must take place over a secure
+//! channel that satisfies mutual entity authentication and integrity.
```
Comment (indomitableSwan) [inline src/lib.rs:96]: Nit: I would delete "secure" and say "... satisfies sender authentication and message integrity". Then delete L106 and move up L107-112.

File: src/lib.rs [hunk 3]
```diff
@@ -88,6 +88,37 @@
...
+//!
+//! The protocol requires a UC-secure, synchronous, authenticated broadcast
+//! channel for use by the [`Participant`]s. Currently, the library handles this
+//! automatically by implementing the echo-broadcast protocol described by
+//! Goldwasser and Lindell[^echo]. This is the approach the paper recommends,
+//! but we note that this reduces the security of the protocol to selective
+//! abort[^abort] rather than the stronger notion of identifiable abort that a
+//! perfect authenticated broadcast protocol would allow.
+//!
+//! All messages sent between participants require mutual entity authentication.
```
Comment (indomitableSwan) [inline src/lib.rs:106]: ie delete me

File: src/lib.rs [hunk 4]
```diff
@@ -88,6 +88,37 @@
...
+//! In practice, this is typically instantiated using a public key
+//! infrastructure (PKI) to associate signing keys to entities.
+//! Briefly, each message is accompanied by a signature on that message, under
+//! the signing key associated with the sending entity.
+//! Since the library does not deal directly with communication channels, it
+//! cannot validate that messages are correctly associated with their sender.
```
Comment (indomitableSwan) [inline src/lib.rs:112]: move me up and change "... cannot validate that messages ..." to "... does not validate that messages ..."

File: src/lib.rs [hunk 5]
```diff
@@ -88,6 +88,37 @@
...
+//!
+//! The calling application is responsible for sending messages between
+//! participants in the protocol.
+//! All communication between [`Participant`]s must take place over a secure
+//! channel that satisfies mutual entity authentication and integrity.
+//!
+//! The protocol requires a UC-secure, synchronous, authenticated broadcast
+//! channel for use by the [`Participant`]s. Currently, the library handles this
+//! automatically by implementing the echo-broadcast protocol described by
+//! Goldwasser and Lindell[^echo]. This is the approach the paper recommends,
```
Comment (indomitableSwan) [inline src/lib.rs:101]: nit: switch "recommends" to "mentions"

File: src/lib.rs [hunk 6]
```diff
@@ -88,6 +88,37 @@
...
+//! but we note that this reduces the security of the protocol to selective
+//! abort[^abort] rather than the stronger notion of identifiable abort that a
+//! perfect authenticated broadcast protocol would allow.
```
Comment (indomitableSwan) [inline src/lib.rs:104]: nits: 
- "note" ---> "emphasize" (or something stronger than "note")
-  delete "perfect", e.g., reword to "... that is achieved with an authenticated broadcast protocol"

**Review by indomitableSwan — CHANGES_REQUESTED** [review]
Mostly nits for clarity

File: src/lib.rs (hunk 1 above)
Comment (marsella) [reply src/lib.rs:52]: I rephrased these lines but I don't want this numbered list to have one massive bullet point. I like having this as the overview section with pointers to details. I moved the networking section up so it's immediately following, though.

**Comment by marsella** [comment]
Thanks for the feedback, this is ready for re-review.

**Review by indomitableSwan — APPROVED** [review]
LGTM

**Pull request merged**